guppylang/engine.py

```diff
@@ -196,6 +196,7 @@ def compile(self, id: DefId) -> ModulePointer:
         # Prepare Hugr for this module
         graph = hf.Module()
         graph.metadata["name"] = "__main__"
+        graph.metadata["guppy_version"] = guppylang.__version__
```

Comment:
We should use a generic field like 
```suggestion
        graph.metadata["generator"] = f"guppylang@{guppylang.__version__}"
```
so the hugr tooling can report it with error messages etc.